textdb/textdb-dataflow/src/test/java/edu/uci/ics/textdb/dataflow/keywordmatch/KeywordMatcherTest.java

```diff
@@ -0,0 +1,402 @@
+package edu.uci.ics.textdb.dataflow.keywordmatch;
+
+import edu.uci.ics.textdb.api.common.*;
+import edu.uci.ics.textdb.api.storage.IDataReader;
+import edu.uci.ics.textdb.api.storage.IDataWriter;
+import edu.uci.ics.textdb.common.constants.LuceneConstants;
+import edu.uci.ics.textdb.common.constants.SchemaConstants;
+import edu.uci.ics.textdb.common.constants.TestConstants;
+import edu.uci.ics.textdb.common.exception.DataFlowException;
+import edu.uci.ics.textdb.common.field.*;
+import edu.uci.ics.textdb.dataflow.common.KeywordPredicate;
+import edu.uci.ics.textdb.dataflow.source.IndexSearchSourceOperator;
+import edu.uci.ics.textdb.dataflow.utils.TestUtils;
+import edu.uci.ics.textdb.storage.LuceneDataStore;
+import edu.uci.ics.textdb.storage.reader.LuceneDataReader;
+import edu.uci.ics.textdb.storage.writer.LuceneDataWriter;
+import org.apache.lucene.analysis.Analyzer;
+import org.apache.lucene.analysis.TokenStream;
+import org.apache.lucene.analysis.standard.StandardAnalyzer;
+import org.apache.lucene.analysis.tokenattributes.CharTermAttribute;
+import org.apache.lucene.queryparser.classic.MultiFieldQueryParser;
+import org.apache.lucene.queryparser.classic.ParseException;
+import org.apache.lucene.queryparser.classic.QueryParser;
+import org.apache.lucene.search.BooleanClause;
+import org.apache.lucene.search.BooleanQuery;
+import org.apache.lucene.search.Query;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.StringReader;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+
+/**
+ * @author Prakul
+ *
+ */
+
+public class KeywordMatcherTest {
+
+    private KeywordMatcher keywordMatcher;
+    private IDataWriter dataWriter;
+    private LuceneDataStore dataStore;
+    private IndexSearchSourceOperator indexSearchSourceOperator;
+    private Analyzer analyzer;
+    private Query queryObj;
+    private Schema schema;
+
+    @Before
+    public void setUp() throws Exception {
+        dataStore = new LuceneDataStore(LuceneConstants.INDEX_DIR, TestConstants.SCHEMA_PEOPLE);
+        analyzer = new StandardAnalyzer();
+        dataWriter = new LuceneDataWriter(dataStore, analyzer);
+        dataWriter.clearData();
+        dataWriter.writeData(TestConstants.getSamplePeopleTuples());
+        schema = dataStore.getSchema();
+    }
+
+    @After
+    public void cleanUp() throws Exception {
+        dataWriter.clearData();
+    }
+
+    /**
+     * Tokenizes the query string using the given analyser
+     * @param analyzer
+     * @param query
+     * @return ArrayList<String> list of results
+     */
+    public ArrayList<String> tokenizeQuery(Analyzer analyzer,  String query) {
```

Comment:
I feel this function can be used in other places as well.  If you agree, can you move it to the Utils package as a helper function?